aries/aries_vcx/src/protocols/did_exchange/state_machine/requester/request_sent/mod.rs

```diff
@@ -134,3 +130,96 @@ impl DidExchangeRequester<RequestSent> {
+            return Err(AriesVcxError::from_msg(
+                AriesVcxErrorKind::InvalidInput,
+                "DIDExchange response did not have a valid DIDDoc signature from the expected \
+                 inviter",
+            ));
+        }
+
+        let did_doc = attachment_to_diddoc(did_doc_attach)?;
+        if did_doc.id().to_string() != their_did {
+            return Err(AriesVcxError::from_msg(
+                AriesVcxErrorKind::InvalidInput,
+                "DIDExchange response had a DIDDoc which did not match the response DID",
+            ));
+        }
+        return Ok(did_doc);
+    }
+
+    if let Some(did_rotate_attach) = response.content.did_rotate {
+        debug!(
+            "Verifying signature of DID Rotate attached to response: {did_rotate_attach:?} \
+             against expected key {invitation_key:?}"
+        );
+        let verified_signature =
+            jws_verify_attachment(&did_rotate_attach, invitation_key, wallet).await?;
+        if !verified_signature {
+            return Err(AriesVcxError::from_msg(
+                AriesVcxErrorKind::InvalidInput,
+                "DIDExchange response did not have a valid DID rotate signature from the expected \
+                 inviter",
+            ));
+        }
+
+        let AttachmentType::Base64(signed_did_b64) = did_rotate_attach.data.content else {
+            return Err(AriesVcxError::from_msg(
+                AriesVcxErrorKind::EncodeError,
+                "DIDExchange response did not have a valid DID rotate attachment",
+            ));
+        };
+
+        let did_bytes = URL_SAFE_LENIENT.decode(signed_did_b64).map_err(|_| {
+            AriesVcxError::from_msg(
+                AriesVcxErrorKind::EncodeError,
+                "DIDExchange response did not have a valid base64 did rotate attachment",
+            )
+        })?;
+        let signed_did = String::from_utf8(did_bytes).map_err(|_| {
+            AriesVcxError::from_msg(
+                AriesVcxErrorKind::EncodeError,
+                "DIDExchange response did not have a valid UTF8 did rotate attachment",
+            )
+        })?;
+
+        if signed_did != their_did {
+            return Err(AriesVcxError::from_msg(
+                AriesVcxErrorKind::InvalidInput,
+                format!(
+                    "DIDExchange response had a DID rotate which did not match the response DID. \
+                     Wanted {their_did}, found {signed_did}"
+                ),
+            ));
+        }
+
+        let did = &Did::parse(their_did)?;
+        let DidResolutionOutput {
+            did_document: did_doc,
+            ..
+        } = resolver_registry.resolve(did, &Default::default()).await?;
+        return Ok(did_doc);
+    }
+
+    // default to error
+    Err(AriesVcxError::from_msg(
+        AriesVcxErrorKind::InvalidInput,
+        "DIDExchange response could not be verified. No DIDDoc nor DIDRotate was attached.",
+    ))
```

Comment:
Do we have a strategy in Aries VCX as how to abort a protocol and send a problem report? As I would think this would be an appropriate time to create a problem report.

Reply:
I recognize that's somewhat outside the scope of this PR, but it came to mind while reviewing

Reply:
Hmmm, so this protocol state machine is a little different to some of the other state machines in aries-vcx (this one is the "most modern"), but the handler has a `.fail` method and `.receive_problem_report` method which creates and processes problem report messages: https://github.com/hyperledger/aries-vcx/blob/main/aries/aries_vcx/src/protocols/did_exchange/state_machine/mod.rs#L37

I think the thinking was that, you get returned a TransitionError, you should process the error and decide if it's a fatal error or not; and subsequently decide if you want to create a problem report with `.fail`, or just try again

Reply:
it isn't being tested or demonstrated anywhere though.. so i think in general it's been a bit neglected

Reply:
Got it. I'd say that perhaps it's something we should look at at some point, but would be outside the scope of this PR.